File: CommonUtils/src/hu/fx/config/ConfigReader.scala
package hu.fx.config

import java.io.File
import java.io.FileInputStream
import java.util.Properties

class ConfigReader(path: String) {

  private val properties = ConfigReader.readProperties()
  
  def getProperty(key: String): Option[String] = {
    val value = properties.getProperty(key)
    if (value == null) None
    else Some(value)
  }

  private object ConfigReader {
    def readProperties() = {
      val props = new Properties
      val loaded = props.load(new FileInputStream(new File(path.format(EnvironmentSupplier.getEnvironment()))))
      props
    }
  }
}

File: PresentationService/src/main/java/hu/fx/presentation/config/ConfigReader.java
package hu.fx.presentation.config;

import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

public class ConfigReader {

	private Properties props = new Properties();
	
	private void load() throws IOException {
		InputStream inStream = this.getClass().getClassLoader().getResourceAsStream("WEB-INF/config.properties");
		props.load(inStream);
	}
	
	public ConfigReader() {
		try {
			load();
		} catch (IOException e) {
			e.printStackTrace();
			props = null;
		}
	}
	
	public String getParam(String key) {
		if (props != null) {
			return (String) props.get(key);
		}
		
		return "";
	}
	
}
